syntax/verilog_systemverilog.vim

```diff
@@ -312,6 +312,14 @@ if index(s:verilog_syntax_fold, "define") >= 0 || index(s:verilog_syntax_fold, "
     " verilogGlobal keywords. This syntax match fixes that:
     syn match verilogGlobal "\<`\(ifn\?def\|e\(els\(e\|if\)\)\|ndif\)\>"
 endif
+if index(s:verilog_syntax_fold, "instance") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
+    syn region verilogFold
+        \ start="^\s*\w\+\(\s*#\s*(\|\s\+\w\+\s*(\)\(.*)\s*\w\+\s*;\)\@!"
```

Comment:
This is similar to what you said, but it confirms there is no `;` at the end of the line instead of searching for `(` explicitly. This allows all the examples I added in the test below.